graphics/diff-pdf/Portfile

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                diff-pdf
+version             0.3
+categories          graphics
+maintainers         {@pietvo vanoostrum.org:piet}
+platforms           darwin
+supported_archs     i386 x86_64
+license             GPL-2
+
+description         diff-pdf is a tool for visually comparing two PDFs.
+long_description \
+    It takes two PDF files as arguments. By default, its only output is \
+    its return code, which is 0 if there are no differences and 1 if the \
+    two PDFs differ. If given the `--output-diff` option, it produces a \
+    PDF file with visually highlighted differences. \
+    With the `--view` argument it opens a window that lets you view the \
+    files' pages and zoom in on details.
+       
+homepage            http://vslavik.github.io/diff-pdf
+
+PortGroup           github 1.0
+github.setup        vslavik diff-pdf 0.3 v
+github.tarball_from releases
+
+PortGroup           wxWidgets 1.0
+wxWidgets.use       wxWidgets-3.0
+
+checksums           rmd160  938e1f32218a083c2db37e8c220bbde48ef091e7 \
+                    sha256  4e91f1094216b98baa22d461b9662b3b6666e49ceb01671e5db6d1a3d384c538 \
+                    size    33688
+
+depends_lib         port:poppler \
+                    port:cairo \
```

Comment:
Can cairo-devel also be used? If so, change this to:

`path:lib/libcairo.dylib:cairo
`

Reply:
I am going to check that.

Reply:
I have deactivated cairo and installed cairo-devel, changed the dependency line to the suggestion above, but it will not compile. Here is part of the log file that talks about cairo.
I also tried the popper way: "path:lib/pkgconfig/cairo.pc:cairo". 
I even tried to remove the explicit cairo dependency, as it is already a dependency of popper, but that doesn't make a difference. So I think the answer is NO. Unless someone shows me a way around.

:debug:main Found Dependency: path: /opt/local/lib filename: libcairo.dylib regex: ^libcairo.dylib$
:debug:main epoch: in tree: 0 installed: 0
:debug:main cairo-devel 1.17.2_0 exists in the ports tree
:debug:main cairo-devel 1.17.2_0 +quartz+x11 is the latest installed
:debug:main cairo-devel 1.17.2_0 +quartz+x11 is active
:debug:main Merging existing variants '+quartz+x11' into variants
:debug:main new fully merged portvariants: x11 + quartz +
:debug:main Changing to port directory: /opt/local/var/macports/sources/rsync.macports.org/release/tarballs/ports/graphics/cairo-devel

...

:debug:main Found Dependency: path: /opt/local/lib/pkgconfig filename: cairo.pc regex: ^cairo.pc$

:debug:main Found Dependency: path: /opt/local/lib filename: libcairo.dylib regex: ^libcairo.dylib$
:debug:main Searching for dependency: cairo
:debug:main Didn't find receipt, going to depspec regex for: cairo
:debug:main Found Dependency: path: /opt/local/lib filename: libcairo.dylib regex: ^libcairo.dylib$
...

:debug:configure Found Dependency: path: /opt/local/lib filename: libcairo.dylib regex: ^libcairo.dylib$
:debug:configure Registry error: cairo not registered as installed & active.
:debug:configure     invoked from within
:debug:configure "registry_active $portname"
:debug:configure     (procedure "recursive_collect_deps" line 4)
:debug:configure     invoked from within
:debug:configure "recursive_collect_deps $name $depsfound"
:debug:configure     (procedure "recursive_collect_deps" line 12)
:debug:configure     invoked from within
:debug:configure "recursive_collect_deps $name $deplist"
:debug:configure     (procedure "target_run" line 141)
:debug:configure     invoked from within
:debug:configure "{*}$handler $ditem"
:error:configure Registry error: cairo not registered as installed & active.

Reply:
That does not look like a build error, so I think you might be doing something wrong. However, the procedure you mentioned is the correct way to test it. 
 - Change dependency in the Portfile to `path:lib/libcairo.dylib:cairo`
 - `sudo port deactivate cairo`
 - `sudo port install cairo-devel`
 - `sudo port clean diff-pdf`
 - `sudo port install diff-pdf`

Reply:
Also, rebuild the Portindex first with `port sync`.

Reply:
port sync did the trick.